libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2298,7 +2298,12 @@ MAV_RESULT GCS_MAVLINK::handle_command_set_message_interval(const mavlink_comman
     const int32_t interval_us = (int32_t)packet.param2;
 
     uint16_t interval_ms;
-    if (interval_us == 0) {
+
+    if (msg_id == MAVLINK_MSG_ID_HEARTBEAT && (interval_us <= 1000000)) {
```

Comment:
This structure means you can never reset the heartbeat to its default rate.  Also, some links might really, really not want heartbeat and we should allow that.  I'm thinking sattelite high-latency-high-cost.

Reply:
Satellite needs you to be able to persistently control the messaging rates, or you are going to burn a lot of money on turning stuff off on boot....

Reply:
not supported, for now a reset will set to 1Hz. If we want the value to be persistent, all values should be set to that and that's work for a new PR.